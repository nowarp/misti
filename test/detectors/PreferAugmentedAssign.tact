contract Test {
    field: Int = 19;
    fun test() {
        let a: Int = 42;

        // Should be reported:
        a = a + 1;
        a = 1 + a;
        self.field = 1 + self.field;
        self.field = self.field + a;

        // Complex logic; should not be reported:
        self.field = self.field + a + 1;
        self.field = -self.field + a;
    }
}
