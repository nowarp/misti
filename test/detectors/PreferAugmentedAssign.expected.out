[INFO] PreferAugmentedAssign: Prefer augmented assignment: a += 1
test/detectors/PreferAugmentedAssign.tact:7:9:
  6 |         // Should be reported:
> 7 |         a = a + 1;
              ^
  8 |         a = 1 + a;
Help: Consider using augmented assignment instead: a += 1
See: https://nowarp.io/tools/misti/docs/detectors/PreferAugmentedAssign

[INFO] PreferAugmentedAssign: Prefer augmented assignment: a += 1
test/detectors/PreferAugmentedAssign.tact:8:9:
  7 |         a = a + 1;
> 8 |         a = 1 + a;
              ^
  9 |         self.field = 1 + self.field;
Help: Consider using augmented assignment instead: a += 1
See: https://nowarp.io/tools/misti/docs/detectors/PreferAugmentedAssign

[INFO] PreferAugmentedAssign: Prefer augmented assignment: self.field += 1
test/detectors/PreferAugmentedAssign.tact:9:9:
   8 |         a = 1 + a;
>  9 |         self.field = 1 + self.field;
               ^
  10 |         self.field = self.field + a;
Help: Consider using augmented assignment instead: self.field += 1
See: https://nowarp.io/tools/misti/docs/detectors/PreferAugmentedAssign

[INFO] PreferAugmentedAssign: Prefer augmented assignment: self.field += a
test/detectors/PreferAugmentedAssign.tact:10:9:
   9 |         self.field = 1 + self.field;
> 10 |         self.field = self.field + a;
               ^
  11 | 
Help: Consider using augmented assignment instead: self.field += a
See: https://nowarp.io/tools/misti/docs/detectors/PreferAugmentedAssign